### PR TITLE
Bugfix for multipart requests

### PR DESCRIPTION
**Pull request by dluc**
# Type of change? 

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

When uploading a file via the proxy, the application throws an exception because the Content-Type header is handled incorrectly. The service also hides the exception returning a 200 OK.

This change fixes the content type management, and also ensures internal errors are surfaced correctly in case of bugs, i.e. not returning 200 OK.

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/reverse-proxy-dotnet/27)

### CODE REVIEW

File: ProxyAgent/HttpClient/HttpRequest.cs [hunk 1]
```diff
@@ -153,8 +155,21 @@ public IHttpRequest SetContent<T>(T sourceObject, Encoding encoding, MediaTypeHe
...
+
+            // When content type is not defined, then assume JSON because that's the typical use for this application.
+            // If the content type is defined, and it's not a file transfer, then set the content type.
+            // For file transfers (via multipart), the content type is set in HttpClient.SetHeaders
+            // using TryAddWithoutValidation. Setting multipart content type here would cause an exception.
+            if (mediaType == null)
+            {
+                this.ContentType = this.defaultMediaType;
+            }
+            else if (!mediaType.StartsWith(MULTIPART_REQUEST_CONTENT_TYPE_PREFIX, StringComparison.InvariantCultureIgnoreCase))
```
Comment (elvinm) [inline ProxyAgent/HttpClient/HttpRequest.cs]: Where are you setting the content type when it is a multipart request?

Comment (dluc) [reply ProxyAgent/HttpClient/HttpRequest.cs]: in `HttpClient.cs` (see above comment on line 161)

Comment (elvinm) [reply ProxyAgent/HttpClient/HttpRequest.cs]: I saw the comment but as you said in the comment it says if the content type is defined and not a file transfer. So when it is a file transfer (multipart/form-data) you don't set content type to anything in this logic from what I can tell. Is that expected and works okay?

Comment (dluc) [reply ProxyAgent/HttpClient/HttpRequest.cs]: yes it works, we ran multiple tests with different http request types

**Pull request merged**